Gem/Code/Source/Components/NetworkTeleportComponent.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "NetworkTeleportComponent.h"
+#include <Multiplayer/Components/NetworkTransformComponent.h>
+
+#include <AzCore/Component/TransformBus.h>
+#include <AzCore/Component/ComponentApplicationBus.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+#include <AzFramework/Physics/RigidBodyBus.h>
+
+namespace MultiplayerSample
+{
+	void NetworkTeleportComponent::Reflect(AZ::ReflectContext* reflection)
+	{
+		if (auto serializationContext = azrtti_cast<AZ::SerializeContext*>(reflection))
+		{
+			serializationContext->Class<NetworkTeleportComponent, Component>()
+				->Field("Destination", &NetworkTeleportComponent::m_reset)
+				->Version(3);
+
+			if (AZ::EditContext* editContext = serializationContext->GetEditContext())
+			{
+				editContext->Class<NetworkTeleportComponent>("Network Teleport Component",
+					"Teleports colliding player entities to a fixed location")
+					->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+					->Attribute(AZ::Edit::Attributes::Category, 
+						"MultiplayerSample")
+					->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, 
+						AZ_CRC_CE("Game"))
+					->DataElement(0, &NetworkTeleportComponent::m_reset,
+						"Destination", "Entity at which to teleport the colliding player")
+					;
+			}
+		}
+	}
+
+	NetworkTeleportComponent::NetworkTeleportComponent()
+		: m_trigger([this](
+			AzPhysics::SceneHandle,
+			const AzPhysics::TriggerEventList& eventList)
+			{
+				OnTriggerEvents(eventList);
+			})
+	{
+	}
+
+	void NetworkTeleportComponent::Activate()
+	{
+		auto* si = AZ::Interface<AzPhysics::SceneInterface>::Get();
+		if (si != nullptr)
+		{
+			AzPhysics::SceneHandle sh = si->GetSceneHandle(
+				AzPhysics::DefaultPhysicsSceneName);
+			si->RegisterSceneTriggersEventHandler(sh, m_trigger);
```

Comment:
This must be an example from my book. Since then I found a way to detect the trigger without singing up for ALL triggers. Here is how to sign for triggers on an entity that has a rigid body (this wont work on static colliders!):

1. Inside Activate()
```
        AzPhysics::RigidBody* body = nullptr;
        Physics::RigidBodyRequestBus::EventResult(body, GetEntityId(), &Physics::RigidBodyRequests::GetRigidBody);
        if (body)
        {
            body->RegisterOnCollisionBeginHandler(m_collisionHandler);
        }
```
2. Define an event:
```
        void OnCollisionBegin(AzPhysics::SimulatedBodyHandle body, const AzPhysics::CollisionEvent& collisionEvent);
        AzPhysics::SimulatedBodyEvents::OnCollisionBegin::Handler m_collisionHandler{ [this](
            AzPhysics::SimulatedBodyHandle body, const AzPhysics::CollisionEvent& collisionEvent)
        {
            OnCollisionBegin(body, collisionEvent);
        } };
```

3. Handle the event without having to loop over all collision events.
```
void YourComponent::OnCollisionBegin(
        AzPhysics::SimulatedBodyHandle body,
        const AzPhysics::CollisionEvent& collisionEvent)
    {
        if (collisionEvent.m_body2)
        {
             ....
```

Reply:
>This must be an example from my book

Most of this component is a rip off of your book :P (with a sprinkle of modifications (all bad))